### DISCUSSION TITLE
The ability to "Auto change" page per streamdeck in multi Streamdecks (XL) setup

### DISCUSSION THREAD

**Discussion by MikeLemo1** [StreamController/StreamController#140] [Ideas]
I'm messing around with the auto change feature inside the SC pages window (Which feels quite heavy to have to open it like that in a new window besides it being slow to open and laggy with the presses BTW) using 2 StreamDeck XL (32 buttons) I might have messed something but I'm unable to set auto page switch to work for each individual SD seperatelly they just switch both to the said SC page in linux app switch where I want to define this behavior per each SD.
on the way also wondering if switch pages per press mode (mainly "While switch hold" but also ON/OFF toggle option) is available and supported in StreamController?

**Comment by Core447** [comment]
Unfortunately deck specific automatic page switches are currently not supported. However, this feature is already implemented in the code and will be available with the next release coming in the next weeks. The coming version will also allow you specify when actions should trigger (on hold, on short press etc) as well as adding states which can be used to create toggle buttons.

 
> laggy with the presses BTW)

What exactly is laggy? The ui of the page manager? If yes, is this also the case for the main window?

**Reply by MikeLemo1** [reply]
Not sure what you mean by main window if that's the buttun matrix setup panel then it's fine animations are a bit choppy on my machine but mostly the open page window and changing between page tabs takes around half a second between each movement which is a bit unpleasant but generally it's fine

**Reply by Core447** [reply]
Yeah, I mean the "normal" window with the button grid.
I'll look into this and come back to you

**Reply by v4u6h4n** [reply]
I got some lagginess when I was really pushing my CPU, but was otherwise fine the rest of the time, so seemed to only be hardware lag.
